### ISSUE TITLE
Fail to configure with Android NDK r22

### ISSUE THREAD

**Issue by wongsyrone** [openssl/openssl#13685]
No configdata.pm generated.

OpenSSL Version: 1.1.1i

Analysis:

The condition on https://github.com/openssl/openssl/blob/OpenSSL_1_1_1i/Configurations/15-android.conf#L33 doesn't meet anymore. The new NDK doesn't have `platforms` folder and `AndroidVersion.txt`.

```
$ ./Configure android-x86 no-shared                                                                                                                                                                      
Configuring OpenSSL version 1.1.1i (0x1010109fL) for android-x86
Using os-specific seed configuration
$ANDROID_NDK_HOME=/opt/android-ndk is invalid at (eval 10) line 37.
```

```
$ export
ANDROID_NDK=/opt/android-ndk
ANDROID_NDK_HOME=/opt/android-ndk
COLORFGBG='15;0'
COLORTERM=truecolor
DBUS_SESSION_BUS_ADDRESS='unix:path=/run/user/1000/bus'
DESKTOP_SESSION=/usr/share/xsessions/plasma
DISPLAY=:0
EDITOR=vim
GIT_ASKPASS=/usr/bin/ksshaskpass
GOPATH=/home/wong/go
GRADLE_HOME=/usr/share/java/gradle
GTK2_RC_FILES=/etc/gtk-2.0/gtkrc:/home/wong/.gtkrc-2.0:/home/wong/.config/gtkrc-2.0
GTK_IM_MODULE=fcitx
GTK_MODULES=canberra-gtk-module
GTK_RC_FILES=/etc/gtk/gtkrc:/home/wong/.gtkrc:/home/wong/.config/gtkrc
HISTFILE=/home/wong/.zhistory
HISTSIZE=1000
HOME=/home/wong
KDED_STARTED_BY_KDEINIT=1
KDE_APPLICATIONS_AS_SCOPE=1
KDE_FULL_SESSION=true
KDE_SESSION_UID=1000
KDE_SESSION_VERSION=5
KONSOLE_DBUS_SERVICE=:1.147
KONSOLE_DBUS_SESSION=/Sessions/1
KONSOLE_DBUS_WINDOW=/Windows/1
KONSOLE_VERSION=201200
LANG=en_US.UTF-8
LANGUAGE=''
LOGNAME=wong
LS_COLORS='rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:'
MAIL=/var/spool/mail/wong
MOTD_SHOWN=pam
MOZ_PLUGIN_PATH=/usr/lib/mozilla/plugins
OLDPWD=/home/wong
PAM_KWALLET5_LOGIN=/run/user/1000/kwallet5.socket
PATH=/usr/lib/ccache/bin/:/home/wong/go/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/android-ndk:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
PROFILEHOME=''
PWD=/home/wong
QT_AUTO_SCREEN_SCALE_FACTOR=0
QT_IM_MODULE=fcitx
SAVEHIST=1000
SHELL=/usr/bin/zsh
SHELL_SESSION_ID=73674d586cb2440eb775f6e1829f212d
SHLVL=1
TERM=xterm-256color
USER=wong
WINDOWID=123731975
XAUTHORITY=/home/wong/.Xauthority
XCURSOR_SIZE=24
XCURSOR_THEME=breeze_cursors
XDG_CURRENT_DESKTOP=KDE
XDG_RUNTIME_DIR=/run/user/1000
XDG_SEAT=seat0
XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0
XDG_SESSION_CLASS=user
XDG_SESSION_DESKTOP=KDE
XDG_SESSION_ID=2
XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session1
XDG_SESSION_TYPE=x11
XDG_VTNR=7
XMODIFIERS='@im=fcitx'
XZ_OPT='--threads=0'
ZLSCOLORS='rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:'
```

**Comment by petrovr** [comment]
Syrone Wong wrote:
> No configdata.pm generated.
>
> OpenSSL Version: 1.1.1i
>
> Analysis:
>
> The condition on https://github.com/openssl/openssl/blob/OpenSSL_1_1_1i/Configurations/15-android.conf#L33 doesn't meet anymore. The new NDK doesn't have `platforms` folder and `AndroidVersion.txt`.

But you could use https://android-developers.googleblog.com/2019/10/introducing-ndk-r21-our-first-long-term.html instead new betas.

Regards
Roumen Petrov

**Comment by wongsyrone** [comment]
> But you could use https://android-developers.googleblog.com/2019/10/introducing-ndk-r21-our-first-long-term.html instead new betas. Regards Roumen Petrov

1. It's a bug that should be fixed instead encourage users to use old versions
2. r22 is a **stable** release instead of beta, r21 is just an ordinary release with extended support
3. I use Archlinux, the NDK version is out of my control. I have to maintain my own PKGBUILD to keep using  NDK r21, it's user-unfriendly.
4. The fix is easy to apply, the original author has already sent the PR to backport the fix to the 1.1.1 branch, I fixed it on my side.

**Comment by wongsyrone** [comment]
Close due to https://github.com/openssl/openssl/pull/13694
